lib/Support/ProgramRunner.cpp

```diff
@@ -0,0 +1,38 @@
+#include "llvm/ADT/SmallVector.h"
+#include "llvm/Support/FileSystem.h"
+#include "llvm/Support/Path.h"
+#include "llvm/Support/Program.h"
+
+#include "revng/Support/Assert.h"
+#include "revng/Support/ProgramRunner.h"
+
+using namespace llvm;
+using namespace sys;
+
+ProgramRunner Runner;
+
+ProgramRunner::ProgramRunner() :
+  MainExecutable(fs::getMainExecutable("", reinterpret_cast<void *>(main))) {
+  Paths = { { path::parent_path(MainExecutable) } };
+
+  // Append PATH
+  char *Path = getenv("PATH");
+  if (Path == nullptr)
+    return;
+  StringRef(Path).split(Paths, ":");
+}
+
+void ProgramRunner::run(llvm::StringRef ProgramName,
```

Comment:
Isn't this going to be used by people that shouldn't assert? Shall we return an `llvm::Error`?

Reply:
it seems that the underlyin api returns
```
  /// -1 indicates failure to execute
  /// -2 indicates a crash during execution or timeout
  /// everything else is the return code of the invoked program
```
i guess assert that is positive and return a error if not zero?

Reply:
Up to you, we could even return error even if < 0.

Reply:
i will assert that the return code is not -1 or -2 to avoid an potential attacker to figure out that they have some control over an external program invocation